### DISCUSSION TITLE
Volt Language selector

### DISCUSSION THREAD

**Discussion by maicol07** [livewire/livewire#8780] [1. Help] [answered]
Hi,
how can I refresh all the components in the page without doing a full refresh using Volt? In my case, I switch the active language via a select and I'd like to change all the texts when the new language is selected from the select.
Thanks

Selector example:
```php
<?php
use Livewire\Volt\state;
state('locale');

$setLocale = function () {
	session()->put('locale', $locale)
	app()->setLocale($locale)
	// There is a middleware that does setLocale if locale is in session
}
?>
<select id="language-selector" wire:model="locale">
	<option value="en">English</option>
	<option value="fr">French</option>
	<option value="de">German</option>
	<option value="it">Italian</option>
</select>
```

**Comment by EdgarsJoja** [comment, marked as answer]
Have you tried using `wire:navigate`? It would essentially replace all HTML without doing the page reload.
https://livewire.laravel.com/docs/navigate#redirects

**Reply by maicol07** [reply]
Works! Thanks!
Ended up udating the set_locale function like this:
```php
$setLocale = function () {
	session()->put('locale', $locale)
	app()->setLocale($locale)
	// There is a middleware that does setLocale if locale is in session
    $this->redirect(url()->previous(), navigate: true);
}
```

**Reply by jeremycharbonnel** [reply]
Hey @maicol07, Would you mind posting the full solution? I am trying something very similar and can't make it work. How do you use wire:navigate in this context? Thank you!

**Reply by maicol07** [reply]
Hi @jeremycharbonnel, the code above is everything that has changed from the code in the OP. The only thing I've changed is the `$this->redirect(url()->previous(), navigate: true);` and that did the trick!

**Reply by jeremycharbonnel** [reply]
@maicol07 Thank you for the fast answer! That is very strange then because I have the exact same code and it does not do anything at all apart from refreshing the page ...

**Reply by maicol07** [reply]
Have you set the navigate parameter at the end?

**Reply by jeremycharbonnel** [reply]
Hey @maicol07, the code in the volt component is the same and it reloads the page but it does not take account of the session locale. Would you mind sharing your middleware to see the difference? Thanks!

**Reply by maicol07** [reply]
Oh, yeah without the middleware it doesn't work:
```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;

class LocaleMiddleware
{
    public function handle(Request $request, Closure $next): Response
    {
        $locale = session('locale', 'en');
        app()->setLocale($locale);
        return $next($request);
    }
}
```

Basically it only retrieves the locale from the session and sets it with `setLocale`

**Reply by jeremycharbonnel** [reply]
@maicol07 Thanks! My problem seemed to be that my middleware was declared as global which apparently does not work with session. I had to declare to declare it for the "web" specifically!

Now it works for me, but I was wondering if the "app()->setLocale($locale)" is useful in the Volt component as the middleware handles this?

In any case, thank you!

**Reply by maicol07** [reply]
Well, in almost all cases it can be ignored (I can remove it too), it's useful only when you do something that requires translation on the same request (error messages, ...).
But for 99% of the situations and how it's written the Volt component the `setLocale` can be safely removed!
